builtin/providers/aws/resource_aws_instance.go

```diff
@@ -296,6 +311,17 @@ func resourceAwsInstanceCreate(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
+	if v := d.Get("vpc_security_group_ids"); v != nil {
+		for _, v := range v.(*schema.Set).List() {
+			str := v.(string)
+
+			var g ec2.SecurityGroup
+			g.Id = str
+
+			runOpts.SecurityGroups = append(runOpts.SecurityGroups, g)
```

Comment:
should this be `runOpts.SecurityGroupIDs` ?
`SecurityGroups` is a collection of Security Group names.